sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -369,17 +390,23 @@ public UnboundedJmsReader(
 
     @Override
     public boolean start() throws IOException {
-      ConnectionFactory connectionFactory = source.spec.getConnectionFactory();
+      Read spec = source.spec;
+      ConnectionFactory connectionFactory = spec.getConnectionFactory();
       try {
-        this.connection = connectionFactory.createConnection();
+        if (spec.getUsername() != null) {
+          this.connection =
+              connectionFactory.createConnection(spec.getUsername(), spec.getPassword());
```

Comment:
Not clear to me from the [Javadoc](https://docs.oracle.com/javaee/7/api/javax/jms/ConnectionFactory.html#createConnection-java.lang.String-java.lang.String-) whether password is allowed to be `null`. Can you check? If so, you might want to try `MoreObjects.firstNonNull(password, "");`